### PR TITLE
Feature selecting time interval

### PR DESCRIPTION
**Pull request by AliceDmitrieva**
https://trello.com/c/WylCYTPf/2-showing-scrolling-horizontal-time-list
https://trello.com/c/1PoDgNQ0/3-selecting-time

### CODE REVIEW

File: app/src/main/java/com/storytellingsoftware/dmitrieva/retrocorder/MainActivity.java [hunk 1]
```diff
@@ -10,22 +10,31 @@
...
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+        LinearLayoutManager layoutManager = new LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false);
+        RecyclerView recyclerView = findViewById(R.id.recyclerView);
+        recyclerView.setLayoutManager(layoutManager);
+
+        TimeIntervalAdapter timeIntervalAdapter = new TimeIntervalAdapter(getResources().getIntArray(R.array.timeIntervals), this);
+        recyclerView.setAdapter(timeIntervalAdapter);
+        recyclerView.getBackground().setAlpha(80);
```
Comment (oxied) [inline app/src/main/java/com/storytellingsoftware/dmitrieva/retrocorder/MainActivity.java]: It's better to use alfa channel in color.xml

File: app/src/main/java/com/storytellingsoftware/dmitrieva/retrocorder/MainActivity.java [hunk 2]
```diff
@@ -35,7 +44,7 @@ public void onCreate(Bundle savedInstanceState) {
 
     private void openCamera() {
         camera = getCameraInstance(0);
-        preview = new Preview(this, camera);
+        Preview preview = new Preview(this, camera);
```
Comment (oxied) [inline app/src/main/java/com/storytellingsoftware/dmitrieva/retrocorder/MainActivity.java]: Got an exception in emulator (api 22):
```
Process: com.storytellingsoftware.dmitrieva.retrocorder, PID: 3634
    java.lang.NullPointerException: Attempt to invoke virtual method 'void android.hardware.Camera.setPreviewDisplay(android.view.SurfaceHolder)' on a null object reference
        at com.storytellingsoftware.dmitrieva.retrocorder.Preview.surfaceCreated(Preview.java:27)
        at android.view.SurfaceView.updateWindow(SurfaceView.java:580)
        at android.view.SurfaceView$3.onPreDraw(SurfaceView.java:176)
        at android.view.ViewTreeObserver.dispatchOnPreDraw(ViewTreeObserver.java:944)
        at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1970)
        at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1061)
        at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:5885)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:767)
        at android.view.Choreographer.doCallbacks(Choreographer.java:580)
        at android.view.Choreographer.doFrame(Choreographer.java:550)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:753)
        at android.os.Handler.handleCallback(Handler.java:739)
        at android.os.Handler.dispatchMessage(Handler.java:95)
        at android.os.Looper.loop(Looper.java:135)
        at android.app.ActivityThread.main(ActivityThread.java:5254)
        at java.lang.reflect.Method.invoke(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:372)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
```

I rotated emulator to landscape.

File: app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -1,12 +1,21 @@
 <?xml version="1.0" encoding="utf-8"?>
-<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    android:layout_width="fill_parent"
-    android:layout_height="fill_parent"
-    android:orientation="horizontal">
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical">
```
Comment (oxied) [inline app/src/main/res/layout/activity_main.xml]: For which layouts orientation is available/make sense to use?

File: app/src/main/res/layout/list_item.xml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="wrap_content"
+    android:layout_height="match_parent"
+    android:orientation="vertical">
+
+    <TextView
+        android:id="@+id/number"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="@dimen/margin_beth_time_intervals"
```
Comment (oxied) [inline app/src/main/res/layout/list_item.xml]: beth - what does it mean?

File: app/src/main/res/values/arrays.xml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+        <item>4</item>
+        <item>5</item>
+        <item>6</item>
+        <item>7</item>
+        <item>8</item>
+        <item>9</item>
+        <item>10</item>
+    </integer-array>
+
+</resources>
```
Comment (oxied) [inline app/src/main/res/values/arrays.xml]: Add new line

File: app/src/main/res/values/dimens.xml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <dimen name="margin_beth_time_intervals">25dp</dimen>
+    <dimen name="margin_from_time_interval">5dp</dimen>
+    <dimen name="time_interval_text_size">25sp</dimen>
```
Comment (oxied) [inline app/src/main/res/values/dimens.xml]: Material design uses numbers multiple of 8.

File: app/src/main/java/com/storytellingsoftware/dmitrieva/retrocorder/MainActivity.java (hunk 1 above)
Comment (AliceDmitrieva) [reply app/src/main/java/com/storytellingsoftware/dmitrieva/retrocorder/MainActivity.java]: Fixed in branch feature_selecting_time_interval.

File: app/src/main/res/layout/list_item.xml (hunk 4 above)
Comment (AliceDmitrieva) [reply app/src/main/res/layout/list_item.xml]: It means "between".

File: app/src/main/res/layout/activity_main.xml (hunk 3 above)
Comment (AliceDmitrieva) [reply app/src/main/res/layout/activity_main.xml]: For LinearLayout. 
Fixed.

File: app/src/main/res/values/arrays.xml (hunk 5 above)
Comment (AliceDmitrieva) [reply app/src/main/res/values/arrays.xml]: Fixed.

File: app/src/main/res/values/dimens.xml (hunk 6 above)
Comment (AliceDmitrieva) [reply app/src/main/res/values/dimens.xml]: Fixed.

File: app/src/main/java/com/storytellingsoftware/dmitrieva/retrocorder/MainActivity.java (hunk 2 above)
Comment (AliceDmitrieva) [reply app/src/main/java/com/storytellingsoftware/dmitrieva/retrocorder/MainActivity.java]: Was added to tasks, will be fixed in next pull requests.

File: app/src/main/java/com/storytellingsoftware/dmitrieva/retrocorder/MainActivity.java (hunk 1 above)
Comment (oxied) [reply app/src/main/java/com/storytellingsoftware/dmitrieva/retrocorder/MainActivity.java]: OK

File: app/src/main/java/com/storytellingsoftware/dmitrieva/retrocorder/MainActivity.java (hunk 2 above)
Comment (oxied) [reply app/src/main/java/com/storytellingsoftware/dmitrieva/retrocorder/MainActivity.java]: OK

File: app/src/main/res/layout/activity_main.xml (hunk 3 above)
Comment (oxied) [reply app/src/main/res/layout/activity_main.xml]: Good.

File: app/src/main/res/layout/list_item.xml (hunk 4 above)
Comment (oxied) [reply app/src/main/res/layout/list_item.xml]: OK

File: app/src/main/res/values/arrays.xml (hunk 5 above)
Comment (oxied) [reply app/src/main/res/values/arrays.xml]: ok

File: app/src/main/res/values/dimens.xml (hunk 6 above)
Comment (oxied) [reply app/src/main/res/values/dimens.xml]: ok

**Comment by oxied** [comment]
Looks good.

**Comment by AliceDmitrieva** [comment]
Merge pull request #2 from AliceDmitrieva/feature_selecting_time_interval

**Comment by AliceDmitrieva** [comment]
Merge pull request #2 from AliceDmitrieva/feature_selecting_time_interval

**Pull request merged**